subprojects/plugins/src/main/java/org/gradle/api/plugins/JvmTestSuitePlugin.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.plugins;
+
+import org.gradle.api.ExtensiblePolymorphicDomainObjectContainer;
+import org.gradle.api.Incubating;
+import org.gradle.api.NamedDomainObjectProvider;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.plugins.jvm.JUnitPlatformTestingFramework;
+import org.gradle.api.plugins.jvm.JvmTestSuite;
+import org.gradle.api.plugins.jvm.JvmTestingFramework;
+import org.gradle.api.plugins.jvm.internal.DefaultJvmTestSuite;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.testing.Test;
+import org.gradle.platform.base.TestSuite;
+import org.gradle.platform.base.plugins.TestingExtension;
+
+/**
+ * <p>A {@link org.gradle.api.Plugin} which allows for defining, compiling and running groups of Java tests against (potentially)
+ * various different target environments.</p>
+ *
+ * @see <a href="https://docs.gradle.org/current/userguide/test_suite_plugin.html">Test Suite plugin reference</a>
+ *
+ * @since 7.3
+ */
+@Incubating
+public class JvmTestSuitePlugin implements Plugin<Project> {
+    public static final String DEFAULT_TEST_SUITE_NAME = SourceSet.TEST_SOURCE_SET_NAME;
+
+    @Override
+    public void apply(Project project) {
+        project.getPluginManager().apply("org.gradle.test-suite-base");
+        project.getPluginManager().apply("org.gradle.java-base");
```

Comment:
Why do we apply by id instead of `apply(Class<?>)`?

Reply:
It makes the plugins report in build scans prettier because you get the human readable id vs just the class name.

Reply:
I was wondering why these were switched as well.  I guess all sorts of tests would fail if the string didn't match an actual plugin, but wouldn't it be better to call a static method on the Plugin class that would read this string from the properties file to keep the names aligned and fail at compile time in case it changes?  

Though I guess the names wouldn't change all that frequently, if ever.

Reply:
Right, this name wouldn't change once we've decided what it was unless we added a new alias.